src/core/tokenStore.js

```diff
@@ -204,56 +194,48 @@ export default function tokenStoreFactory(options) {
          * (Could happen if maxSize is reduced during the life of the tokenStore)
          * @returns {Promise} - resolves when done
          */
-        enforceMaxSize: function enforceMaxSize() {
-            var self = this;
-            return this.getIndex().then(function(latestIndex) {
-                var keysToRemove;
-                var excess = latestIndex.length - config.maxSize;
-                if (excess > 0) {
-                    keysToRemove = latestIndex.slice(0, excess);
-                    return Promise.all(
-                        _.map(keysToRemove, function(key) {
-                            return self.remove(key);
-                        })
-                    );
-                }
-                return true;
-            });
+        enforceMaxSize() {
+            return this.getIndex()
+                .then(latestIndex => {
+                    const excess = latestIndex.length - config.maxSize;
+                    if (excess > 0) {
+                        const keysToRemove = latestIndex.slice(0, excess);
+                        return Promise.all(
+                            _.map(keysToRemove, key => this.remove(key))
+                        );
+                    }
+                    return true;
+                });
         },
 
         /**
-         * Checks one token and removes it from the store if expired
-         * @param {Object} token - the token object
+         * Checks one token and removes it from the store if expired.
+         * If the timeLimit is lesser than or equal to 0, no time limit is applied.
+         * @param {token} token - the token object
          * @returns {Promise<Boolean>}
          */
-        checkExpiry: function checkExpiry(token) {
-            if (Date.now() - token.receivedAt > config.tokenTimeLimit) {
+        checkExpiry(token) {
+            const {tokenTimeLimit} = config;
+            if (tokenTimeLimit > 0 && Date.now() - token.receivedAt > tokenTimeLimit) {
                 return this.remove(token.value);
             }
-            return true;
+            return Promise.resolve(true);
         },
 
         /**
          * Checks all the tokens in the store to see if they expired
          * @returns {Promise<Boolean>} - resolves to true
          */
-        expireOldTokens: function expireOldTokens() {
-            var self = this;
-            return self.getTokens().then(function(tokens) {
+        expireOldTokens() {
+            return this.getTokens()
                 // Check each token's expiry, synchronously:
-                return _.reduce(
+                .then(tokens => _.reduce(
```

Comment:
Could we use ES6 Array.reduce() in place of lodash _.reduce() ?

Reply:
The same here, `tokens` is not an array